tools/KoreBuild.Console/Commands/DockerBuildCommand.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class DockerBuildCommand : SubCommandBase
+    {
+        private const string DockerfileExtension = ".dockerfile";
+        private const string ImagePrefix = "aspnetbuild";
+
+        public CommandArgument Platform { get; set; }
+
+        public List<string> Arguments {get; set; }
+        
+        public string ContainerName { get; set; } = "testcontainer";
+
+        public override void Configure(CommandLineApplication application)
+        {
+            Platform = application.Argument("platform", "The docker platform to run on.");
+            Arguments = application.RemainingArguments;
+
+            base.Configure(application);
+        }
+
+        protected override bool IsValid()
+        {
+            if(string.IsNullOrEmpty(Platform?.Value))
+            {
+                Reporter.Error("Platform is a required argument.");
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override int Execute()
+        {
+            var dockerFileSource = GetDockerFileSource(Platform.Value);
+            var dockerFileDestination = Path.Combine(RepoPath, GetDockerFileName(Platform.Value));
+
+            File.Copy(dockerFileSource, dockerFileDestination, true);
+
+            // If our ToolSource isn't http copy it to the docker context
+            var dockerToolsSource = ToolsSource;
+            string toolsSourceDestination = null;
+            if (!ToolsSource.StartsWith("http"))
+            {
+                dockerToolsSource = "ToolsSource";
+                toolsSourceDestination = Path.Combine(RepoPath, dockerToolsSource);
+                DirectoryCopy(ToolsSource, toolsSourceDestination);
+            }
+
+            try
+            {
+                var buildArgs = new List<string> { "build" };
+
+                var tag = $"{ImagePrefix}:{ContainerName}";
+                buildArgs.AddRange(new string[] { "-t", tag, "-f", dockerFileDestination, RepoPath });
+                var buildResult = RunDockerCommand(buildArgs);
+
+                if (buildResult != 0)
+                {
+                    return buildResult;
+                }
+
+                var runArgs = new List<string> { "run", "--rm", "-it", "--name", ContainerName, tag };
```

Comment:
If we're using `--name` we should use a uniquely-generated (possibly timestamp-based?) name, perhaps something like `aspnetbuild_[timestamp]`. If there's already a container with this name, it will conflict.

Also, we should make it possible to disable `--rm`. If my build fails, it would be nice to have the container stick around.

Finally, make sure the container name is dumped to the console so I know which containers come from which builds.

Reply:
`--name` is most useful for long-running containers.  In this scenario, I would just use the random name assigned by Docker.

Reply:
I figured you wouldn't ever run two at the same time due to `--rm`, but I guess it's still possible, and as below I would like to let you run interactive in the future, so I'll do that.

I had intended to add disabling `--rm` as an option later on. I was intending to get this running in an actual scenario (one of the CI builds) before complicating things with more options. I just filed #395 for that. Is that good or do you think that option is more important that that?

Good point about the container name.

Reply:
I think `--name` becomes relevant when `--rm` is removable. @mikeharder is definitely right that if the container is just going to be removed after the build, there's no need to know the name.

Perhaps the best solution is to remove `--name` for now, and when we make `--rm` removable, we can also add an option to the `docker-build` command to allow providing a container name.

Reply:
I'm actually partial to supplying the name ourselves because:
1. We can easily output the name in case it persists for some reason (including future interactivity).
2. It makes it easy to tell at a glance which containers belong to KoreBuild.
3. It doesn't hurt anything (unless there's something I don't know).

I do agree that allowing people to specify the container name in the future would be a nice feature. I created #396 to enumerate all the future improvements we think of while doing this work.